### ISSUE TITLE
[Feature Request] Allow `shopifyDomain` that does not include trailing `myshopify.com`

### ISSUE THREAD

**Issue by jplhomer** [Shopify/storefront-kit#97]
**Is your feature request related to a problem? Please describe.**
Today, we hardcode a trailing `myshopify.com` to the developer's storefront domain when creating the storefront client: https://github.com/Shopify/hydrogen-ui/blob/2022-10/packages/react/src/storefront-client.ts#L44-L48

This is because it represents the "permanent domain" for a Shopify storefront, and because it's the most commonly used format.

However, technically _any_ domain pointed at the storefront can field API requests, not just the permanent domain. 

Specifically, this unlocks:

- A potential future where a session could be established via cookie when a client-side request is made from the same domain
- Better internal tooling (pointing a Hydrogen v2 installation at a Spin instance of core, for example)

**Describe the solution you'd like**
The shop domain should respect whatever value I provide to it.

**Comment by frehner** [comment]
Hmm, so, what would the ideal API for this look like? 

1. An optional second param to not include `.myshopify.com`?
    ```ts
    getShopifyDomain({domain: 'my-domain.com', omitMyShopify: true}) // my-domain.com
    ```
2. Revert the decision to add `.myshopify.com` by default?
    ```ts
    getShopifyDomain({domain: 'my-domain.com'}) // my-domain.com
    getShopifyDomain({domain: 'my-domain'}) // my-domain (and you have to add .myshopify.com yourself)
    getShopifyDomain({domain: 'my-domain.myshopify.com'}) // my-domain.myshopify.com
    ```
3. Some other idea?

Thoughts @jplhomer ?

**Comment by jplhomer** [comment]
@frehner Oooh yes I like it. Probably go with `#1` for now (non-breaking), and switch to `#2` in a future breaking change.

**Comment by frehner** [comment]
Well, I'm going through the breaking changes right now, so I may go straight for #2 haha

**Comment by frehner** [comment]
Resolved and released in `2023.1.1`
